mailpoet/lib/AdminPages/Pages/DynamicSegments.php

```diff
@@ -143,7 +143,7 @@ public function render() {
           ]
         );
 
-        if (!isset($attributeTerms['errors'])) {
+        if ((!$attributeTerms instanceof \WP_Error) && !isset($attributeTerms['errors'])) {
```

Comment:
Nitpick: Would it be better to flip the condition and check for the existence of the fields we expect? Check if the `product_attributes` are present in the `$data` array and so on?